src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -2957,99 +2956,40 @@ Real64 GLHESlinky::calcHXResistance(EnergyPlusData &state)
 
 //******************************************************************************
 
-Real64 GLHEBase::interpGFunc(Real64 const LnTTsVal // The value of LN(t/TimeSS) that a g-function
-) const
+Real64 GLHEBase::interpGFunc(Real64 const LnTTsVal) const
 {
-    // SUBROUTINE INFORMATION:
-    //       AUTHOR         Chris L. Marshall, Jeffrey D. Spitler
-    //       DATE WRITTEN   1993
-    //       MODIFIED       August, 2000
-    //       RE-ENGINEERED Dan Fisher
+    // Purpose: interpolate between g-function values, with linear extrapolation above and below range
 
-    // PURPOSE OF THIS SUBROUTINE:
-    //    To interpolate or extrapolate data in GFILE
-    //    to find the correct g-function value for a
-    //    known value of the natural log of (T/Ts)
+    auto &x = this->myRespFactors->LNTTS;
+    auto &y = this->myRespFactors->GFNC;
```

Comment:
Normally, using references to lists is not the right approach, but this is a bit different.  In this case, you don't actually know which index you are going to use yet.  Using `x` and `y` here make this logic **extremely** easy to read and understand.  :+1: